File: CS/E4811/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace E4811.Controllers {
    public class HomeController : Controller {
        [HttpGet]
        public ActionResult Index() {            
            return View(new SearchOrder() { 
                EmployeeID = 1,
                DateFrom= new DateTime(1995, 5, 1),
                DateTo = new DateTime(1996, 2, 29)
            });
        }

        [HttpPost]
        public ActionResult Index(SearchOrder model) {
            return View(model);
        }

        public ActionResult OrdersGridPartial(int employeeID, DateTime dateFrom, DateTime dateTo) {
            SearchOrder search = new SearchOrder() {
                EmployeeID = employeeID,
                DateFrom = dateFrom,
                DateTo = dateTo
            };
            ViewData["SearchOrder"] = search;
            return PartialView("_OrdersGridPartial", Order.SearchOrders(search));
        }

        public ActionResult OrderDetailsGridPartial(int orderID) {
            ViewData["OrderID"] = orderID;
            return PartialView("_OrderDetailsGridPartial", OrderDetail.GetOrderDetails(orderID));
        }
    }
}

File: VB/E4811/Controllers/HomeController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc


Namespace E4811.Controllers
	Public Class HomeController
		Inherits Controller

		<HttpGet>
		Public Function Index() As ActionResult
			Return View(New SearchOrder() With {
				.EmployeeID = 1,
				.DateFrom= New Date(1995, 5, 1),
				.DateTo = New Date(1996, 2, 29)
			})
		End Function

		<HttpPost>
		Public Function Index(ByVal model As SearchOrder) As ActionResult
			Return View(model)
		End Function

		Public Function OrdersGridPartial(ByVal employeeID As Integer, ByVal dateFrom As Date, ByVal dateTo As Date) As ActionResult
			Dim search As New SearchOrder() With {
				.EmployeeID = employeeID,
				.DateFrom = dateFrom,
				.DateTo = dateTo
			}
			ViewData("SearchOrder") = search
			Return PartialView("_OrdersGridPartial", Order.SearchOrders(search))
		End Function

		Public Function OrderDetailsGridPartial(ByVal orderID As Integer) As ActionResult
			ViewData("OrderID") = orderID
			Return PartialView("_OrderDetailsGridPartial", OrderDetail.GetOrderDetails(orderID))
		End Function
	End Class
End Namespace